hazelcast/src/test/java/com/hazelcast/map/impl/mapstore/writebehind/TransactionsWithWriteBehind_whenNoCoalescingQueueIsFullTest.java

```diff
@@ -0,0 +1,430 @@
+import org.junit.Test;
+import org.junit.experimental.categories.Category;
+import org.junit.rules.ExpectedException;
+import org.junit.runner.RunWith;
+
+import java.util.Collection;
+import java.util.Collections;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.atomic.AtomicBoolean;
+
+import static com.hazelcast.map.impl.mapstore.writebehind.WriteBehindFlushTest.assertWriteBehindQueuesEmpty;
+import static com.hazelcast.transaction.TransactionOptions.TransactionType.ONE_PHASE;
+import static com.hazelcast.transaction.TransactionOptions.TransactionType.TWO_PHASE;
+import static java.util.concurrent.TimeUnit.SECONDS;
+import static org.hamcrest.core.Is.isA;
+import static org.junit.Assert.assertEquals;
+
+@RunWith(HazelcastParallelClassRunner.class)
+@Category({QuickTest.class, ParallelJVMTest.class})
+public class TransactionsWithWriteBehind_whenNoCoalescingQueueIsFullTest extends HazelcastTestSupport {
+
+    @Rule
+    public ExpectedException expectedException = ExpectedException.none();
+
+    @Test
+    public void prepare_step_throws_reached_max_size_exception_when_two_phase() {
+        expectedException.expect(TransactionException.class);
+        expectedException.expectCause(isA(ReachedMaxSizeException.class));
+
+        String mapName = "map";
+        long maxWbqCapacity = 100;
+
+        Config config = getConfig(mapName, maxWbqCapacity);
+        HazelcastInstance node = createHazelcastInstance(config);
+
+        TransactionContext context = newTransactionContext(node, TWO_PHASE);
+        context.beginTransaction();
+
+        TransactionalMap map = context.getMap(mapName);
+
+        for (int i = 0; i < 101; i++) {
+            map.put("item-" + i, "value");
+        }
+        context.commitTransaction();
+    }
+
+    private static TransactionContext newTransactionContext(HazelcastInstance node,
+                                                            TransactionOptions.TransactionType twoPhase) {
+        TransactionOptions options = new TransactionOptions()
+                .setTransactionType(twoPhase);
+        return node.newTransactionContext(options);
+    }
+
+    @Test
+    public void commit_step_does_not_throw_reached_max_size_exception_when_two_phase() {
+        String mapName = "map";
+        long maxWbqCapacity = 10;
+        int keySpace = 100;
+
+        Config config = getConfig(mapName, maxWbqCapacity);
+        HazelcastInstance node = createHazelcastInstance(config);
+
+        TransactionContext context = newTransactionContext(node, TWO_PHASE);
+        context.beginTransaction();
+
+        TransactionalMap txMap = context.getMap(mapName);
+
+        for (int i = 0; i < keySpace; i++) {
+            txMap.remove("item-" + i);
+        }
+
+        try {
+            context.commitTransaction();
+        } catch (TransactionException e) {
+            context.rollbackTransaction();
+        }
+
+        sleepSeconds(10);
```

Comment:
Can we make a comment why we need this sleep here?

Reply:
It was left over. Removed it.